api/hostdb.go

```diff
@@ -0,0 +1,116 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+type (
+	// ExtendedHostDBEntry is an extension to modules.HostDBEntry that includes
+	// the string representation of the public key, otherwise presented as two
+	// fields, a string and a base64 encoded byte slice.
+	ExtendedHostDBEntry struct {
+		modules.HostDBEntry
+		PublicKeyString string `json:"publickeystring"`
```

Comment:
yeah the problem with JSON is not the api, it's that the frontend needs some way to let the user pick out hosts individually. That could be done by giving each host their own button, and I'm sure some people will do it that way, but for example in siac you definitely cannot give each host a button. You'd have to give them a number and then somehow know that the number corresponded to something exact in the... well it just doesn't work in a stateless app like siac.

I'm not happy that we sometimes present public keys in the json as base64 and now sometimes as strings, but I definitely feel that strings is the right move, and wish we had started there. Just like we chose for addresses to be hex strings.